[case fastenum_user_methods]

from fastenum import Enum

class Color(Enum):
    RED = 1
    GREEN = 2
    BLUE = 3

    @property
    def prop(self) -> bool:
        return True

    def is_true(self) -> bool:
        return False



reveal_type(Color('RED')) # E: Revealed type is 'main.Color'
reveal_type(Color('RED').prop) # E: Revealed type is 'builtins.bool'
reveal_type(Color('RED').is_true()) # E: Revealed type is 'builtins.bool'

[/CASE]
