# Test various ways how to import fastenum
# as mypy module is registered based on name used in AST

[case fastenum_import_from]

from fastenum import Enum

class Color(Enum):
    RED = 1
    GREEN = 2
    BLUE = 3

reveal_type(Color.RED) # E: Revealed type is 'main.Color'
reveal_type(Color.GREEN) # E: Revealed type is 'main.Color'
reveal_type(Color.GREEN.value) # E: Revealed type is 'Any'

[/CASE]


[case fastenum_import_full]

import fastenum

class Color(fastenum.Enum):
    RED = 1
    GREEN = 2
    BLUE = 3

reveal_type(Color.RED) # E: Revealed type is 'main.Color'
reveal_type(Color.GREEN) # E: Revealed type is 'main.Color'
reveal_type(Color.GREEN.value) # E: Revealed type is 'Any'

[/CASE]
